libs/domains/organizations/feature/src/lib/helm-repository-create-edit-modal/helm-repository-create-edit-modal.tsx

```diff
@@ -90,8 +103,20 @@ export function HelmRepositoryCreateEditModal({
       return
     }
 
-    // Omit `login_type` in the request
-    const { login_type, ...config } = helmRepositoryRequest.config
+    // Omit `login_type` and `aws_auth_type` in the request (they're local UI state)
+    const { login_type, aws_auth_type, ...config } = helmRepositoryRequest.config
+
+    // For OCI_ECR STATIC auth, remove role_arn if present
+    // For OCI_ECR STS auth, remove access_key_id and secret_access_key
+    if (helmRepositoryRequest.kind === 'OCI_ECR') {
```

Comment:
Rebuild config cleanly based on aws_auth_type (more elegant than mutating with delete)

```suggestion
const { login_type, aws_auth_type, access_key_id, secret_access_key, role_arn, ...rest } =
  helmRepositoryRequest.config

let finalConfig = rest

if (helmRepositoryRequest.kind === 'OCI_ECR') {
  if (aws_auth_type === 'STATIC') {
    finalConfig = {
      ...rest,
      access_key_id,
      secret_access_key,
    }
  } else if (aws_auth_type === 'STS') {
    finalConfig = {
      ...rest,
      role_arn,
    }
  }
}
```